### DISCUSSION TITLE
Progress of DDLX and Shapes

### DISCUSSION THREAD

**Discussion by balegas** [electric-sql/electric#624] [Feature requests]
This discussion tracks the progress on the two main Electric interfaces, DDLX and Shapes interface, that are available with simplifications and currently under development.

As of the [0.7.0](https://electric-sql.com/blog/2023/11/02/electricsql-v0.7-released) release, it is possible to use DDLX syntax for electrifying tables and the Shapes API is fully supported on the client, but with some (coarse) simplifications: 
- full-table (over)sync: always sync the entire data of a table, independently of the filters you apply
- shapes must include all dependent tables, i.e. a table and all reachable tables following foreign keys.

We'll continue updating this discussion with the progress and examples on how you can use the system at the current state of development.

**Comment by thruflo** [comment]
@balegas would be nice to have an update now shapes are merged.

**Reply by balegas** [reply]
@icehaunter is preparing some docs for the release.

**Reply by maurotrigo** [reply]
Hi there, 
 
 I'm seriously considering Electric to rebuild our mobile and web apps at CityTroops. We've been playing with it for the past 2 weeks, but the current state of the Shape interface really scared us. Since our platform is multi-tenant, over-synchronization is a real problem. Is there a tentative timeline for Shapes? 
 
 Thank you, really love what you're doing.

**Comment by KyleAMathews** [comment]
👋 we've been working the last month on a rebuild of the Electric server over at a temporary repo https://github.com/electric-sql/electric-next/

You can read more about why we made the decision at https://next.electric-sql.com/about

We're really excited about all the new possibilities the new server brings and we hope you'll check it out soon and give us your feedback.

We're now moving the temporary repo back here. As part of that migration we're closing all the old issues, PRs, and discussions. We really appreciate you taking the time to start this discussion! If it's still relevant in the new system, please reopen.
